src/mixins/eraser_brush.mixin.js

```diff
@@ -0,0 +1,653 @@
+          if (obj.erasable && obj.intersectsWithObject(path)) {
+            _this._addPathToObjectEraser(obj, path);
+          }
+        });
+
+        canvas.fire('erasing:end');
+
+        canvas.requestRenderAll();
+        path.setCoords();
+        this._resetShadow();
+
+        // fire event 'path' created
+        canvas.fire('path:created', { path: path });
+      }
+    }
+  );
+
+  /**
+   * EraserPath class
+   * Used by {@link fabric.EraserBrush}
+   * 
+   * {@link fabric.EraserPath} is a workaround for clipping paths that are strokes and not fills.
+   * Clipping is done with the fill of the clip path, so to enable clipping out paths by their stroke and achieving an eraser effect, 
+   * {@link fabric.EraserPath} fills a rect where the object that it needs to clip is drawn.
+   * Then it draws the paths drawn by {@link fabric.EraserBrush} onto the rect in `globalCompositionMode = 'destination-out'`. 
+   * This removes the paths from the drawn rect resulting is a rect that has been erased.
+   * 
+   * {@link fabric.EraserPath} is attached to it's owning object as a clip path. 
+   * Basically it achieves what the `inverted` prop achieves, only for stroked objects.
+   * 
+   * Can be used regardless of {@link fabric.EraserBrush} to create an inverted clip path that contains strokes, unclosed paths or unfilled paths.
+   * 
+   * Without this workaround a clip path containing unclosed paths clips an object as if the path was closed and filled or disregards it.
+   * 
+   * It is possible to provide a clip path to this object, clipping out the drawn rect.
+   * 
+   * @private
+   * @class fabric.EraserPath
+   * @extends fabric.Rect
+   */
+  fabric.EraserPath = fabric.util.createClass(fabric.Rect, fabric.Collection, {
+
+    type: 'eraserPath',
+
+    _objects: [],
+
+    initialize: function (objects, options) {
+      this.callSuper('initialize', Object.assign(options || {}, {
+        originX: 'center',
+        originY: 'center'
+      }));
+      this._objects = objects || [];
+      this._objects.forEach(function (p) {
+        p.path.set({ globalCompositeOperation: 'destination-out' });
+      });
+    },
+
+    /**
+     * Used to set options when erasing
+     * @param {fabric.Object} parent The object that owns this clip path
+     */
+    setParent: function (parent) {
+      this.set({
+        width: parent.width,
+        height: parent.height,
+        clipPath: parent.clipPath
+      });
+    },
+
+    _render: function (ctx) {
+      this.callSuper('_render', ctx);
+      this._objects.forEach(function (o) {
+        ctx.save();
+        var m = o.transformMatrix;
+        m && ctx.transform(m[0], m[1], m[2], m[3], m[4], m[5]);
+        o.path.render(ctx);
+        ctx.restore();
+      });
+    },
+
+    addPath: function (path, transformMatrix) {
+      path.set({ globalCompositeOperation: 'destination-out' });
+      this._objects.push({ path: path, transformMatrix: transformMatrix });
```

Comment:
if the transform matrix is applied to path before it is added then this line is redundant

Reply:
7fd927d